smart-contract-verifier/src/compiler/compilers.rs

```diff
@@ -39,17 +42,19 @@ pub struct Compilers<C> {
     cache: DownloadCache,
     fetcher: Arc<dyn Fetcher>,
     evm_compiler: C,
+    lock: Arc<Semaphore>,
```

Comment:
that's not a lock, it's misleading. lock is usually a single owner.
it's best imo to call semaphores with what kind of resource it manages. in our case it's threads, so we could call this `threads` or something similar, and then it would look like `self.threads.acquire()`

Reply:
will change it to `threads_semaphore`